### ISSUE TITLE
SSL23_GET_SERVER_HELLO error with the new 0.10 release

### ISSUE THREAD

**Issue by dovahcrow** [snapview/tokio-tungstenite#80]
code to reproduce:
```rust
use tokio_tungstenite::connect_async;

#[tokio::main]
async fn main() {
    let url = url::Url::parse(&"wss://www.deribit.com/ws/api/v2").unwrap();

    connect_async(url).await.expect("Failed to connect");
}
```

Compile with `tls` the feature.

Full error
```
thread 'main' panicked at 'Failed to connect: Tls(Ssl(Error { code: ErrorCode(1), cause: Some(Ssl(ErrorStack([Error { code: 336031996, library: "SSL routines", function: "SSL23_GET_SERVER_HELLO", reason: "unknown protocol", file: "s23_clnt.c", line: 794 }]))) }, X509VerifyResult { code: 0, error: "ok" }))', src/libcore/result.rs:1188:5
```

**Comment by dovahcrow** [comment]
Thanks!

**Comment by agalakhov** [comment]
Fixed in 0.10.1, thanks.
